### PR TITLE
DEPR: deprecate .as_blocks()

### LINKED ISSUE #17302: DEPR: .as_blocks and .blocks
These shouldn't be public methods. Deprecate and make private (lead with ``_``) for now.

### PR DESCRIPTION
**Pull request by jreback**
closes #17302

### CODE REVIEW

File: doc/source/10min.rst [hunk 1]
```diff
@@ -95,17 +95,7 @@ will be completed:
...
-   df2.as_matrix          df2.corr
-   df2.astype             df2.corrwith
-   df2.at                 df2.count
-   df2.at_time            df2.cov
-   df2.axes               df2.cummax
-   df2.B                  df2.cummin
-   df2.between_time       df2.cumprod
-   df2.bfill              df2.cumsum
-   df2.blocks             df2.D
+   df2.D
```
Comment (gfyoung) [inline doc/source/10min.rst]: What happened here?

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -488,10 +488,9 @@ Other API Changes
 Deprecations
 ~~~~~~~~~~~~
 - :func:`read_excel()` has deprecated ``sheetname`` in favor of ``sheet_name`` for consistency with ``.to_excel()`` (:issue:`10559`).
-
 - ``pd.options.html.border`` has been deprecated in favor of ``pd.options.display.html.border`` (:issue:`15793`).
-
 - :func:`SeriesGroupBy.nth` has deprecated ``True`` in favor of ``'all'`` for its kwarg ``dropna`` (:issue:`11038`).
+- :func:`DataFrame.as_blocks` is deprecated as this is exposing the internal implementation (:issue:`17302`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: nit: deprecated --> "deprecated,"

File: pandas/core/generic.py [hunk 3]
```diff
@@ -3692,6 +3692,33 @@ def as_blocks(self, copy=True):
...
+        .. deprecated:: 0.21.0
+
+        NOTE: the dtypes of the blocks WILL BE PRESERVED HERE (unlike in
+              as_matrix)
+
+        Parameters
+        ----------
+        copy : boolean, default True
+
+               .. versionadded: 0.16.1
```
Comment (gfyoung) [inline pandas/core/generic.py]: I don't recall: were we still keep `versionadded` that there this old?

**Comment by jreback** [comment]
update, thanks @gfyoung

File: doc/source/10min.rst (hunk 1 above)
Comment (jreback) [reply doc/source/10min.rst]: I took a bunch out rather than regenerate it. ``.as_blocks()`` was in that list

**Pull request merged**